### ISSUE TITLE
Locking field-paths for data consistency

### ISSUE THREAD

**Issue by 0xMochan** [0xPlaygrounds/subgrounds#25]
**Is your feature request related to a problem? Please describe.**
When doing long queries, data gathered towards the end of the query is no longer consistent with the beginning.

**Describe the solution you'd like**
I'd like to ensure data consistency through some form of argument locking.

**Describe alternatives you've considered**
Not making long queries.

**Additional context**
`“Lock” fieldpaths (i.e.: add `"block": {"number": end_block}` and `"where {"_change_block": {"number_gte": start_block}}``
- Basically, locking block ranges so that the data accrued is atomic even if new blocks are indexed come in

**Implementation checklist**
- [ ] Task 1
